contracts/integrations/test/framework/assertions/finalizePool.ts

```diff
@@ -0,0 +1,269 @@
+            expect(events.length, 'Number of RewardsPaid events emitted').to.equal(1);
+            const [rewardsPaidEvent] = events;
+
+            expect(rewardsPaidEvent.poolId, 'RewardsPaid event: poolId').to.equal(poolId);
+            expect(rewardsPaidEvent.epoch, 'RewardsPaid event: currentEpoch_').to.bignumber.equal(currentEpoch);
+
+            const { operatorReward, membersReward } = rewardsPaidEvent;
+            const totalReward = operatorReward.plus(membersReward);
+            assertRoughlyEquals(totalReward, rewards, PRECISION);
+
+            // See _computePoolRewardsSplit
+            if (beforeInfo.poolStats.membersStake.isZero()) {
+                expect(
+                    operatorReward,
+                    "operatorReward should equal totalReward if pool's membersStake is 0",
+                ).to.bignumber.equal(totalReward);
+            } else {
+                expect(operatorReward).to.bignumber.equal(
+                    ReferenceFunctions.getPartialAmountCeil(
+                        new BigNumber(pool.operatorShare),
+                        new BigNumber(stakingConstants.PPM),
+                        totalReward,
+                    ),
+                );
+            }
+
+            // See _syncPoolRewards
+            const expectedTransferEvents = operatorReward.isGreaterThan(0)
+                ? [
+                      {
+                          _from: deployment.staking.stakingProxy.address,
+                          _to: pool.operator,
+                          _value: operatorReward,
+                      },
+                  ]
+                : [];
+
+            // Check for WETH transfer event emitted when paying out operator's reward.
+            verifyEventsFromLogs<WETH9TransferEventArgs>(logs, expectedTransferEvents, WETH9Events.Transfer);
+            // Check that pool rewards have increased.
+            const poolRewards = await stakingWrapper.rewardsByPoolId(poolId).callAsync();
+            expect(poolRewards).to.bignumber.equal(beforeInfo.poolRewards.plus(membersReward));
+            // Check that cumulative rewards have increased.
+            const [
+                mostRecentCumulativeRewards,
+                cumulativeRewardsLastStored,
+            ] = await stakingWrapper.getMostRecentCumulativeReward(poolId).callAsync();
+            expect(cumulativeRewardsLastStored).to.bignumber.equal(currentEpoch);
+            let [numerator, denominator] = ReferenceFunctions.LibFractions.add(
+                beforeInfo.mostRecentCumulativeRewards.numerator,
+                beforeInfo.mostRecentCumulativeRewards.denominator,
+                membersReward,
+                beforeInfo.poolStats.membersStake,
+            );
+            [numerator, denominator] = ReferenceFunctions.LibFractions.normalize(numerator, denominator);
+            // There's a bug in our reference functions, but I can't figure it out :/
```

Comment:
I suspect that the bug is that the reference `safeDiv` truncates in base 10, while solidity truncates in binary

Reply:
🤔 ... I'm down to look into this w you tomorrow!

Reply:
nvm @dorothy-zbornak figured it out, it was a bug in the `LibFractions.normalize` reference function